tools/deps/deps.go

```diff
@@ -0,0 +1,40 @@
+//go:build tools
+
+// This tools/deps package keep track of Go tooling dependency that we use
+// but do not depend directly in code.
```

Comment:
what sorts of tool deps need to be included here? i.e. how did you build this list of deps below? (Is there a straightforward way of automating this?)

Reply:
There are 3 groups added here:

- Go deps we depend on directly in our BUILD files but not our Go files. I.e. `gitlab.com/arm-research/smarter/smarter-device-manager`

- Transitive Go deps that our `http_archive` repo use. For example, in WORKSPACE mode, we used to load some Go deps with
```python
load("@com_github_sluongng_nogo_analyzer//staticcheck:deps.bzl", "staticcheck")

staticcheck()
```

This is no longer possible because of the new way Gazelle operate in Bzlmod

```
ERROR: /Users/sluongng/work/buildbuddy/buildbuddy/BUILD:79:5: //:vet_actual depends on @@org_golang_x_tools//go/analysis/passes/unusedresult:unusedresult in repository @@org_golang_x_tools which failed to fetch. no such package '@@org_golang_x_tools//go/analysis/passes/unusedresult': Unable to load package for @@bazel_gazelle_go_repository_cache//:go.env: The repository '@@bazel_gazelle_go_repository_cache' could not be resolved: Repository '@@bazel_gazelle_go_repository_cache' is not defined
```

- Our transitive proto/grpc deps that we used to add to `server/libmain/libmain.go` to help with `go mod tidy -e` not adding them from `proto/**` generated code.

---

These were pretty much manually picked from the errors I encountered #5792.
So maintaining this list would have to be a manual effort for now.

The UX in Bzlmod is documented here https://github.com/bazelbuild/rules_go/blob/master/docs/go/core/bzlmod.md#depending-on-tools.